packages/workflows/src/api/Properties.ts

```diff
@@ -54,5 +56,53 @@ export class PropertiesWorkflow {
 
         return ZosmfRestClient.getExpectJSON<IWorkflowInfo>(session, resourcesQuery, [Headers.APPLICATION_JSON]);
     }
+
+    /**
+     * Processes the z/OSMF workflow step info
+     * in a recursive manner.
+     *
+     * @protected
+     * @static
+     * @param {IStepInfo[]} steps z/OSMF steps to be processed
+     * @returns {Promise<IStepSummary[]>} Array of z/OSMF step summary objects
+     * @memberof PropertiesWorkflow
+     */
+    public static async processStepSummaries(steps: IStepInfo[]): Promise<IStepSummary[]> {
+        let stepSummaries: IStepSummary[] = [];
+
+        for(const step of steps) {
+            let miscValue: string = "N/A";
+            if(step.submitAs && step.submitAs.match(/.*JCL/)) {
+                if(step.jobInfo && step.jobInfo.jobstatus) {
+                    miscValue = step.jobInfo.jobstatus.jobid;
+                }
+            } else if(step.template) {
+                miscValue = "TSO";
+            } else if(step.isRestStep) {
+                miscValue = `HTTP ${step.actualStatusCode}`;
+            }
+            const stepSummary: IStepSummary = {
```

Comment:
Are you dropping any of the fields from `step` when creating `stepSummary`? You could use the spread operator here and append the `misc` property.